app/controllers/books_controller.rb

```diff
@@ -0,0 +1,12 @@
+class BooksController < ApplicationController
+  include Wor::Paginate
+
+  def index
+    render_paginated Book, each_serializer: BookSerializer
+  end
+
+  def show
+    @book = Book.where(genre: params[:genre], author: params[:author], title: params[:title])
```

Comment:
When you redefine the routes, this endpoint will filter only by the id

Reply:
`.where` method returns an active record relation. This is some kind of array that may contain more than one element. The show endpoint always returns only one element. That's why it should receive the id of the model.

Rails has another method that looks for a record by it's id, and if it can't find it, it raises an exception. This method is `find`. Use that method instead.

http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find